src/h5web/vis-packs/core/utils.ts

```diff
@@ -365,3 +365,30 @@ export function getSliceSelection(
   // Create slice selection string from dim mapping - e.g. [0, 'y', 'x'] => "0,:,:"
   return dimMapping.map((dim) => (isAxis(dim) ? ':' : dim)).join(',');
 }
+
+export function computeSameRatioDomain(
+  availableSize: Size,
+  abscissaDomain: Domain,
+  ordinateDomain: Domain
+): [Domain, Domain] {
+  const { width, height } = availableSize;
+
+  const abscissaRatio = width / (abscissaDomain[1] - abscissaDomain[0]);
+  const ordinateRatio = height / (ordinateDomain[1] - ordinateDomain[0]);
+
+  if (abscissaRatio > ordinateRatio) {
+    const x0 =
+      (abscissaDomain[0] + abscissaDomain[1] - width / ordinateRatio) / 2;
```

Comment:
This formula was derived from solving the system of 2 equations gotten from the following conditions:
- the ratios must be equal
- the center of the canvas must stay the same